File: src/paint_command/draw_line.rs
use super::{Drawable, PaintCommand};
use skia_safe::{Canvas, Color4f, Paint, Path, Point, Rect};

struct DrawLine {
    paint_command: PaintCommand,
    left: f32,
    top: f32,
    right: f32,
    bottom: f32,
    color: Color4f,
    stroke_width: f32,
}

impl DrawLine {
    fn new(
        left: f32,
        top: f32,
        right: f32,
        bottom: f32,
        color: Color4f,
        stroke_width: f32,
    ) -> Self {
        let paint_command = PaintCommand::new(Rect::new(left, top, right, bottom));
        Self {
            paint_command,
            left,
            top,
            right,
            bottom,
            color,
            stroke_width,
        }
    }
}

impl Drawable for DrawLine {
    fn execute(&self, canvas: &Canvas) {
        let mut path = Path::new();
        path.move_to(Point::new(self.left, self.top))
            .line_to(Point::new(self.right, self.bottom));
        let mut paint = Paint::default();
        paint
            .set_stroke_width(self.stroke_width)
            .set_color4f(self.color, None);
        canvas.draw_path(&path, &paint);
    }
}

File: Browser/DrawCommands/DrawLine.cs
using Browser.Utils;
using SkiaSharp;

namespace Browser.DrawCommands;

public sealed class DrawLine : DrawCommand
{
    private readonly string _color;
    private readonly float _thickness;

    public DrawLine(float left,
        float top,
        float right,
        float bottom,
        string color,
        float thickness)
    {
        Rectangle = new SKRect(left, top, right, bottom);
        _color = color;
        _thickness = thickness;
    }

    public override void Draw(SKCanvas canvas)
    {
        var path = new SKPath();
        path.MoveTo(Rectangle.Left, Rectangle.Top);
        path.LineTo(Rectangle.Right, Rectangle.Bottom);
        var paint = new SKPaint
        {
            Color = ColorUtils.ParseColor(_color),
            StrokeWidth = _thickness,
            Style = SKPaintStyle.Stroke
        };
        canvas.DrawPath(path, paint);
    }
}